Dockerfile

```diff
@@ -1,31 +1,77 @@
-FROM php:7.2-fpm
+FROM alpine:latest AS builder
 
-# Install system packages (Git, packages for archives, Node.js)
-RUN apt-get update \
-    && apt-get install -y git libzip-dev zlib1g-dev unzip \
-    && curl -sL https://deb.nodesource.com/setup_14.x | bash - && apt-get install -y nodejs \
-    && rm -rf /var/lib/apt/lists/*
-
-# Install PHP extensions
-RUN docker-php-ext-install pdo_mysql && docker-php-ext-install zip && docker-php-ext-install calendar
-
-# Ensure commands are running successfully before continuing
-SHELL ["/bin/bash", "-o", "pipefail", "-c"]
-
-# Install Composer
-RUN curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/local/bin --filename=composer
+RUN apk add --no-cache --virtual .build-deps \
+    nginx \
+    php7 \
+    php7-fpm \
+    php7-json \
+    php7-session \
+    php7-zip \
+    php7-curl \
+    php7-xml \
+    php7-simplexml \
+    php7-xmlwriter \
+    php7-calendar \
+    php7-fileinfo \
+    php7-tokenizer \
+    php7-dom \
+    php7-pdo_mysql \
+    git \
+    nodejs \
+    npm \
+    zip \
+    composer
 
 # Install Yarn
 RUN npm install -g yarn
 
-WORKDIR /usr/share/nginx/budget
-COPY . .
+# Create and switch to /app work directory and also copy all the necessary files into context
+WORKDIR /app
+COPY . /app
 
 # Required before we're able to run any "php artisan" commands
-RUN composer install
+RUN composer install --no-dev -o
+
+RUN php artisan budget:install && rm .env
+
+FROM alpine:latest
+
+ENV APP_ENV production
+ENV APP_DEBUG false
+
+# Change S6 behaviour so stage when STAGE2 fails the container will exit
+ENV S6_BEHAVIOUR_IF_STAGE2_FAILS 2
+
+# Install all the required runtime dependencies
+RUN apk add --no-cache \
+    nginx \
+    bash \
+    php7 \
+    php7-fpm \
+    php7-json \
+    php7-session \
+    php7-zip \
+    php7-curl \
+    php7-xml \
+    php7-simplexml \
+    php7-xmlwriter \
+    php7-calendar \
+    php7-fileinfo \
+    php7-tokenizer \
+    php7-dom \
+    php7-pdo_mysql
+
+# Install S6 overlay which will, amongst other, do process supervision
+ADD https://github.com/just-containers/s6-overlay/releases/download/v2.0.0.1/s6-overlay-amd64.tar.gz /tmp/
```

Comment:
Don't use add with an URL. Use curl/wget in one run which also deleted the file and ideally compare an hash.